contributingGuides/TS_STYLE.md

```diff
@@ -0,0 +1,517 @@
+
+<a name="ts-nullish-coalescing"></a><a name="1.8"></a>
+
+- [1.8](#ts-nullish-coalescing) **Optional chaining and nullish coalescing**: Use optional chaining and nullish coalescing instead of the `get` lodash function. eslint: [`no-restricted-imports`](https://eslint.org/docs/latest/rules/no-restricted-imports)
+
+  ```ts
+  // BAD
+  import lodashGet from "lodash/get";
+  const name = lodashGet(user, "name", "default name");
+
+  // GOOD
+  const name = user?.name ?? "default name";
+  ```
+
+<a name="type-inference"></a><a name="1.9"></a>
+
+- [1.9](#type-inference) **Type Inference**: When possible, allow the compiler to infer type of variables.
+
+  ```ts
+  // BAD
+  const foo: string = "foo";
+  const [counter, setCounter] = useState<number>(0);
+
+  // GOOD
+  const foo = "foo";
+  const [counter, setCounter] = useState(0);
+  const [username, setUsername] = useState<string | undefined>(undefined); // Username is a union type of string and undefined, and its type cannot be inferred from the default value of undefined
+  ```
+
+  For function return types, default to always typing them unless a function is simple enough to reason about its return type.
+
+  > Why? Explicit return type helps catch errors when implementation of the function changes. It also makes it easy to read code even when TypeScript intellisense is not provided.
+
+  ```ts
+  function simpleFunction(name: string) {
+    return `hello, ${name}`;
+  }
+
+  function complicatedFunction(name: string): boolean {
+    // ... some complex logic here ...
+    return foo;
+  }
+  ```
+
+<a name="jsdoc"></a><a name="1.10"></a>
+
+- [1.10](#jsdoc) **JSDoc**: Omit comments that are redundant with TypeScript. Do not declare types in `@param` or `@return` blocks. Do not write `@implements`, `@enum`, `@private`, `@override`. eslint: [`jsdoc/no-types`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.README/rules/no-types.md)
+
+  ```ts
+  // BAD
+  /**
+   * @param {number} age
+   * @returns {boolean} Whether the person is a legal drinking age or nots
+   */
+  function canDrink(age: number): boolean {
+    return age >= 21;
+  }
+
+  // GOOD
+  /**
+   * @param age
+   * @returns Whether the person is a legal drinking age or nots
+   */
+  ```
+
+<a name="proptypes-and-defaultprops"></a><a name="1.11"></a>
+
+- [1.11](#proptypes-and-defaultprops) **`propTypes` and `defaultProps`**: Do not use them. Use object destructing to assign default values if necessary.
+
+  > Refer to [the propTypes Migration Table](./PROPTYPES_CONVERSION_TABLE.md) on how to type props based on existing `propTypes`.
+
+  ```tsx
+  type MyComponentProps = {
+    requiredProp: string;
+    optionalPropWithDefaultValue?: number;
+    optionalProp?: boolean;
+  };
+
+  function MyComponent({
+    requiredProp,
+    optionalPropWithDefaultValue = 42,
+    optionalProp,
```

Comment:
Should all optional props be required to have a default?

I don't feel strongly either way, but we should clarify that here because currently all optional props are required to have a default.

Reply:
agree we should clarify on this. if you don't pass anything to an optional prop and a default value is not provided, the value for that prop is `undefined`.

If the prop can be `undefined` and the component implementation can handle that situation, then we don't have to require default props to be passed always.

in other words, if the default value is `undefined` (or `null`), default value is not needed imo

I gonna post this in the open source channel as this changes the current guideline

Reply:
Agree with your suggestion